pkg/compact/compact.go

```diff
@@ -309,18 +342,81 @@ func (g *DefaultGrouper) Groups(blocks map[ulid.ULID]*metadata.Meta) (res []*Gro
 			return nil, errors.Wrap(err, "add compaction group")
 		}
 	}
+
+	if g.planner != nil {
```

Comment:
Can planner be .. nil?

Reply:
this will need to be removed, it was put in place because this test breaks and needs to be updated https://github.com/thanos-io/thanos/pull/3807/files#diff-c2110623e0808d8a764bbe216b3a23fa19bf897238848a977810672b0140fb6fR138